src/core/core.layouts.js

```diff
@@ -49,9 +99,152 @@ function findMaxPadding(boxes) {
 }
 
 function addSizeByPosition(boxes, size) {
-	helpers.each(boxes, function(box) {
+	var i, ilen, box;
```

Comment:
Maybe we could remove the side-effect and return the new sizes instead. I think the code will be easier to follow the more we can make it explicit what's being changed and reduce mutability:
```
function addSizeByPosition(size, boxes) {
	var result = extend({}, size);
	var i, ilen, box;
	for (i = 0, ilen = boxes.length; i < ilen; ++i) {
		box = boxes[i].box;
		result[box.position] += box.isHorizontal() ? box.height : box.width;
	}
	return result;
}
```

(I also switched the parameter positions because it's slightly more intuitive to me to have the object we're operating on come first)

I would do the same in the couple methods that call this one. So, for example, now we have:

    updateMinSizes(boxes.outer, params)

I would change to:

    params.outerBoxSizes = calculateMinSizes(boxes.outer, params)